lualib/resty/events/worker.lua

```diff
@@ -266,6 +285,10 @@ function _M:communicate(premature)
         log(ERR, "event worker failed: ", perr)
     end
 
+    wait(read_thread)
+    wait(write_thread)
+    wait(events_thread)
```

Comment:
This seems we are doing duplicate wait as https://github.com/Kong/lua-resty-events/pull/45/files#diff-d0bbfa33cfe00d7863c292aaefa4fb096eeddaa8f0a24539bb90b7ffd56c60c5R269. The wait over there will wake up if one or more of 3 threads terminate.

Reply:
The `wait(x, y, z` means `wait any of them to exit.

While this means "wait all of them to exit":
```
    wait(read_thread)
    wait(write_thread)
    wait(events_thread)
```

Reply:
I see your point. I mean a situation like this:

1. `wait(x, y, z)` returns due to `x` returns.
2. `wait(x)` waits for an already finished thread.

Reply:
@outsinre `wait(x)` on already finished thread will return immediately.

Reply:
@bungle LGTM.